alchemy/src/clickhouse/service.ts

```diff
@@ -0,0 +1,548 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { secret, type Secret } from "../secret.ts";
+import { diff } from "../util/diff.ts";
+import { createClickhouseApi } from "./api.ts";
+import type { ClickhouseClient } from "./api/sdk.gen.ts";
+import type { Service as ApiService, Organization } from "./api/types.gen.ts";
+
+export interface ServiceProps {
+  /**
+   * The key ID for the Clickhouse API
+   */
+  keyId?: string | Secret<string>;
+
+  /**
+   * The secret for the Clickhouse API
+   */
+  secret?: string | Secret<string>;
+
+  /**
+   * The id of Clickhouse cloud organization to create the service in.
+   */
+  organization: string | Organization;
+
+  /**
+   * The name of the Clickhouse service to create.
+   *
+   * @default ${app}-${stage}-${id}
+   */
+  name?: string;
+
+  /**
+   * The underlying cloud provider to create the service on.
+   */
+  provider: ApiService["provider"];
+
+  /**
+   * The region to create the service in.
+   */
+  region: ApiService["region"];
+
+  /**
+   * The IP access list to create the service with.
+   *
+   * @default [{ description: "Anywhere", source: "0.0.0.0/0" }]
+   */
+  ipAccessList?: ApiService["ipAccessList"];
+
+  /**
+   * The minimum replica memory to create the service with.
+   *
+   * @default 8
+   */
+  minReplicaMemoryGb?: ApiService["minReplicaMemoryGb"];
+
+  /**
+   * The maximum replica memory to create the service with.
+   *
+   * @default 356
+   */
+  maxReplicaMemoryGb?: ApiService["maxReplicaMemoryGb"];
```

Comment:
Should we just be doing `extends Omit<ApIService, ..>` instead of duplicating with structural references?